### PR TITLE
Fetch secrets NuGet package from Repo

### LINKED ISSUE #3710: [Infra] Fetch secrets NuGet package from Repox
### Description

The Secrets NuGet package is now owned by the Language Team (code is now in sonar-text).
The NuGet package is now published to https://repox.jfrog.io/repox/api/nuget/sonarsource-nuget-releases.

TODO
- [x] Update NuGet config to fetch secrets from Repox. Might need to trust a new publisher.
- [x] Update package version
- [x] Manually test the new Secrets binary (built from different repo using a different build pipeline)

### PR DESCRIPTION
**Pull request by duncanp-sonar**
Fixes #3710

### CODE REVIEW

**Comment by duncanp-sonar** [comment]
FYI the first commit has the config changes.
The second commit is just the package updates.

File: src/EmbeddedSonarAnalyzer.props [hunk 1]
```diff
@@ -12,7 +12,7 @@
 	     Secrets are special case: we get the analyzer binaries from a NuGet package, and the
 		 rule help from an "normal" Java plugin. The two artefacts are versioned separately.
 	-->
-    <EmbeddedSonarSecretsNuGetVersion>1.0.3</EmbeddedSonarSecretsNuGetVersion>
+    <EmbeddedSonarSecretsNuGetVersion>1.0.4</EmbeddedSonarSecretsNuGetVersion>
     <EmbeddedSonarSecretsJarVersion>2.0.1.611</EmbeddedSonarSecretsJarVersion>
```
Comment (rita-sonarsource) [inline src/EmbeddedSonarAnalyzer.props:16]: The jar is the same I presume?

File: nuget.config [hunk 2]
```diff
@@ -11,15 +11,18 @@
 
   <packageSources>
     <clear />
-    <add key="slvs_input" value="https://sonarsource.pkgs.visualstudio.com/399fb241-ecc7-4802-8697-dcdd01fbb832/_packaging/slvs_input/nuget/v3/index.json" />
+    <!-- SonarSecrets package published by the Language Team -->
+    <add key="repox_releases" value="https://repox.jfrog.io/repox/api/nuget/sonarsource-nuget-releases" />
```
Comment (rita-sonarsource) [inline nuget.config:15]: Are they signed with the same certificate as the dotnet analyzers? If not, the trustedSigners need to be updated.

**Review by rita-sonarsource — APPROVED** [review]
LGTM just the question about the trustedSigners

File: nuget.config (hunk 2 above)
Comment (duncanp-sonar) [reply nuget.config:15]: Sorry, forgot to mention that. The signature is the same. As a test, I commented it out and as expected the NuGet fetch failed.

File: src/EmbeddedSonarAnalyzer.props (hunk 1 above)
Comment (duncanp-sonar) [reply src/EmbeddedSonarAnalyzer.props:16]: Yes. We're now using the latest released versions of the jar and the NuGet package.

**Pull request merged**